File: app/src/main/java/com/example/sasham/testproject/themes/ThemesListingView.java
package com.example.sasham.testproject.themes;

import com.example.sasham.testproject.model.Theme;

import java.util.List;

public interface ThemesListingView {

    void showThemes(List<Theme> themeList);
    void onLoading();
    void onError(String errorMessage);
}

File: app/src/main/java/com/example/sasham/testproject/themes/ThemesListingView.kt
package com.example.sasham.testproject.themes

import com.arellomobile.mvp.MvpView
import com.arellomobile.mvp.viewstate.strategy.AddToEndStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType
import com.example.sasham.testproject.model.Theme

@StateStrategyType(AddToEndStrategy::class)
interface ThemesListingView : MvpView {

    fun showThemes(themeList: List<Theme>)
    fun onLoading()
    fun message(errorMessage: String?)
    fun onThemeClicked(theme: Theme)
}
